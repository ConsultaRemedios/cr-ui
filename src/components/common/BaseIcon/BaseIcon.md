**BaseIcon component** is responsible for rendering cr-ui icons. It uses the
[svg sprite concept](https://fvsch.com/code/svg-icons/) where you define, right after the body tag,
the svg icons with their respective ids (without rendering them) and import them where you want to
(now rendering it).

So, for rendering an icon using the `BaseIcon` component, all you need to do is passing
the reference (id) of the icon you defined previously as the id propriety.
The code would be something like:

#### HTML
```html
<body>
  <svg style="position: absolute; width: 0; height: 0">
    <symbol viewBox="0 0 24 24" id="favorite.icon">
      <path d="M16.4730679,3 C18.034356..."></path>
    </symbol>
  </svg>
  ...
</body>
```

#### JS
```js static
<BaseIcon id="'favorite.icon'">
```

## Using webpack and svg-sprite-loader
Building svg sprites manually is not a good choice. We suggest to use
[svg-sprite-loader](https://github.com/kisenka/svg-sprite-loader) on the projects which helps to
build and get the references of the icons.

Using `webpack` and `svg-sprite-loader`, the same example above,
could be written with just two lines of code - desconsidering the svg file.

```js static
import favorite from 'cr-ui/src/icons/favorite.icon.svg';
<BaseIcon :id="favorite.id">

```

## List of availables icons

```js
<div>
  <div v-for="icon in allIcons" :style="{ display: 'inline-block', padding: '15px 0', width: '33%' }">
    <BaseIcon :id="icon.id"/>
    <span>src/icons/{{icon.id}}.svg</span>
  </div>
</div>
```
