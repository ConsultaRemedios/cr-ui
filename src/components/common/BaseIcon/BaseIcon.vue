<template>
  <svg viewBox="0 0 24 24" :class="$style.icon" v-bind="$attrs">
    <use :xlink:href="`#${id}`" />
  </svg>
</template>

<script>
export default {
  inheritAttrs: false,
  name: 'BaseIcon',
  props: {
    id: String,
  },
};
</script>

<style module>
.icon {
  display: inline-block;
  width: 1em;
  height: 1em;
  vertical-align: middle;
  fill: currentColor;
  font-size: 24px;
}
</style>
