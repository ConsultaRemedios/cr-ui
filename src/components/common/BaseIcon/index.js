import BaseIcon from './BaseIcon.vue';

export default BaseIcon;
