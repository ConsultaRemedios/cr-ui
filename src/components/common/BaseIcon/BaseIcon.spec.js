import { shallowMount } from '@vue/test-utils';
import BaseIcon from './BaseIcon.vue';

describe('BaseIcon component', () => {
  it('matches the snapshot', () => {
    const wrapper = shallowMount(BaseIcon, {
      propsData: {
        id: 'favorite.icon'
      }
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
