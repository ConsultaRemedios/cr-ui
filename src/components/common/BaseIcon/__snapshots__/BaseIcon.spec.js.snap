// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseIcon component matches the snapshot 1`] = `
<svg
  class="icon"
  viewBox="0 0 24 24"
>
  <use
    xlink:href="#favorite.icon"
  />
</svg>
`;
